### PR TITLE
chore: Use signed scripts v0.0.8 for Windows deployments

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:

Contains updates to powershell code used to start kubelet for kube-proxy
 - feat: Updating kubelet/kube-proxy to run with as high priority processes (#4073)
 - feat: Add WinDSR support (#4104)
 - fix: fix that kubeproxy cannot parse featuregates (#4145)

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: .github/workflows/nightly-build.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ name: Nightly
 on:
   schedule:
     - cron: '0 0 * * *' # Every day at midnight
+  workflow_dispatch:
```
Comment (jsturtevant) [inline .github/workflows/nightly-build.yaml:6]: What this do?

Comment (marosset) [reply .github/workflows/nightly-build.yaml:6]: Oops, that wasn't meant to be included in this PR (and has already merged into master)...
It allows one to trigger the pipeline manually or via an API call tho.

Comment (marosset) [reply .github/workflows/nightly-build.yaml]: fixed :)

**Comment by marosset** [comment]
The Windows VHD build is failing because of a regression introduced by #4129 
I'm going to fix that issue in a new PR soon.

**Review by AbelHu — APPROVED** [review]

**Review by AbelHu — CHANGES_REQUESTED** [review]
It seems like that below line in `staging/provisioning/windows/kubeproxystart.ps1` does not work. $args is always empty.
```
$args = ($global:KubeproxyArgList -join " ")
```
when
```
$global:KubeproxyArgList =@("--v=3", "--proxy-mode=kernelspace", "--hostname-override=aksnpwin000000", "--kubeconfig=\config", "--enable-dsr=true", "--feature-gates='WinDSR=true','WinOverlay=false'")
```

**Comment by AbelHu** [comment]
> It seems like that below line in `staging/provisioning/windows/kubeproxystart.ps1` does not work. $args is always empty.
> 
> ```
> $args = ($global:KubeproxyArgList -join " ")
> ```
> 
> when
> 
> ```
> $global:KubeproxyArgList =@("--v=3", "--proxy-mode=kernelspace", "--hostname-override=aksnpwin000000", "--kubeconfig=\config", "--enable-dsr=true", "--feature-gates='WinDSR=true','WinOverlay=false'")
> ```

I tested v0.0.7 in aks with k8s 1.19.1 and it throws an error in kubeproxy.err.log
```
Error: invalid argument "'WinDSR=true','WinOverlay=false'" for "--feature-gates" flag: invalid value of 'WinDSR: true', err: strconv.ParseBool: parsing "true'": invalid syntax
Usage:
  kube-proxy [flags]
```
In C:\k\kubeclusterconfig.json
```
    "Kubernetes":  {
                       "Kubeproxy":  {
                                         "FeatureGates":  [
                                                              "WinDSR=true",
                                                              "WinOverlay=false"
                                                          ]
                                     },
```

It seems like that there is a bug in my previous commit.

**Comment by AbelHu** [comment]
@marosset I tested `examples/windows/kubernetes-windsr.json` with setting below value and there is no error. It seems like to be caused by below three lines.
```
$exe = "$global:KubeDir\kube-proxy.exe"
$args = ($global:KubeproxyArgList -join " ")
[RunProcess.exec]::RunProcess($exe, $args, [System.Diagnostics.ProcessPriorityClass]::AboveNormal)
```

Test data:
"provisioningScriptsPackageURL": "https://testxx3e.blob.core.windows.net/akswindows/signedscripts-v0.0.5.zip"

```
$Global:ClusterConfiguration = ConvertFrom-Json ((Get-Content "c:\k\kubeclusterconfig.json" -ErrorAction Stop) | out-string)
$KubeproxyFeatureGates = $Global:ClusterConfiguration.Kubernetes.Kubeproxy.FeatureGates # This is the initial feature list passed in from aks-engine

$KubeNetwork = "azure"
if ($Global:ClusterConfiguration.Cni.Name -eq "kubenet") {
    $KubeNetwork = "l2bridge"
}

$env:KUBE_NETWORK = $KubeNetwork
$global:HNSModule = "c:\k\hns.psm1"
$global:KubeDir = $Global:ClusterConfiguration.Install.Destination
$global:KubeproxyArgList = @("--v=3", "--proxy-mode=kernelspace", "--hostname-override=$env:computername", "--kubeconfig=$KubeDir\config")

$hnsNetwork = Get-HnsNetwork | ? Name -EQ $KubeNetwork
while (!$hnsNetwork) {
    Write-Host "$(Get-Date -Format o) Waiting for Network [$KubeNetwork] to be created . . ."
    Start-Sleep 10
    $hnsNetwork = Get-HnsNetwork | ? Name -EQ $KubeNetwork
}

# enable WinDsr if WinDsr feature gate is enabled
if ($KubeproxyFeatureGates -contains "WinDSR=true") {
    $global:KubeproxyArgList += @("--enable-dsr=true")
}

if ($KubeproxyFeatureGates.Count -gt 0) {
    $global:KubeproxyArgList += @("--feature-gates='" + ($KubeproxyFeatureGates -join "','") + "'")
}

#
# cleanup the persisted policy lists
#
Import-Module $global:HNSModule
# Workaround for https://github.com/kubernetes/kubernetes/pull/68923 in < 1.14,
# and https://github.com/kubernetes/kubernetes/pull/78612 for <= 1.15
Get-HnsPolicyList | Remove-HnsPolicyList

$KubeproxyCmdline = "$global:KubeDir\kube-proxy.exe "+ ($global:KubeproxyArgList -join " ")
Invoke-Expression $KubeproxyCmdline
```

**Comment by jsturtevant** [comment]
this looks like it is an error with the way the quotes were parsed. note the extra `'`:

```
strconv.ParseBool: parsing "true'":
```

**Comment by AbelHu** [comment]
Created a PR #4145 to fix the issue.

**Comment by jsturtevant** [comment]
the change in #4145 will require a new signed package.

**Comment by marosset** [comment]
v0.0.8 of the signed scripts package includes #4145

**Comment by marosset** [comment]
I set up a cluster using v0.0.8 of the provisioning scripts and set the feature enableWinDSR to true and see kube-proxy running at AboveNormal priority after scheduling some pods.

@AbelHu Do you wan to do any additional validation here or are you OK merging this?

**Comment by AbelHu** [comment]
@marosset, I also used signed scripts v0.0.8 to create an AKS cluster with 1.19.3. The parameters are parsed successfully by kubeproxy in my cluster. Please see the logs as below. And `(Get-HNSpolicylist)[0].Policies.IsDSR` shows `true` after deploying a Windows pod. 

```
I0105 08:12:17.630651    2492 flags.go:59] FLAG: --enable-dsr="true"
....
I0105 08:12:17.707956    2492 feature_gate.go:243] feature gates: &{map[WinDSR:true WinOverlay:false]}
I0105 08:12:17.708661    2492 feature_gate.go:243] feature gates: &{map[WinDSR:true WinOverlay:false]}
```

**Review by AbelHu — APPROVED** [review]

**Pull request merged**